modules/contracts/src.ts/services/ethService.ts

```diff
@@ -368,53 +373,72 @@ export class EthereumChainService extends EthereumChainReader implements IVector
+            }
+          }
+        } else {
           // If an error occurred, throw it to handle it in the catch block.
           const error = result.getError()!;
-          // If the nonce has already been used, the 'original' tx must have been mined and the tx
+          // If the nonce has already been used, one of the original txs must have been mined and the tx
           // we attempted here was a duplicate with bumped gas. Assuming we're on a subsuquent attempt,
-          // handle this by simply proceeding to the confirm block without throwing.
-          if (tryNumber > 1 && error.message.includes("nonce has already been used")) {
+          // handle this by simply proceeding to confirm (each prev tx) without throwing.
+          if (responses.length >= 1 && error.message.includes("nonce has already been used")) {
             // TODO: Confirm that this error message comparison is valid in this context.
             // TODO: Might be better to compare code to NONCE_EXPIRED, etc.
             // A more robust comparison is desirable here either way.
             this.log.info(
               { method, methodId, channelAddress, reason },
-              "Nonce already used: proceeding to check for confirmation.",
+              "Nonce already used: proceeding to check for confirmation in previous transactions.",
             );
-          } else {
-            throw error;
-          }
-        } else {
-          // If response returns undefined, we assume the tx was not sent. This will happen if some logic was
-          // passed into txFn to bail out at the time of sending.
-          let tempResponse = result.getValue();
-          if (!tempResponse) {
-            this.log.warn({ method, methodId, channelAddress, reason }, "Did not attempt tx");
-            // Iff this is the only iteration, then we want to go ahead return w/o saving anything.
-            if (tryNumber === 1) {
-              return Result.ok(undefined);
+            // We must check for confirmation in all previous transactions. Although it's most likely
+            // that it's the previous one, any of them could have been confirmed.
+            for (let i = 0; i < responses.length; i++) {
+              try {
+                receipt = await this.waitForConfirmation(chainId, responses[i]);
+              } catch {}
+              if (receipt) {
+                break;
+              }
             }
           } else {
-            response = tempResponse;
+            throw error;
           }
-
-          // NOTE: Response MUST be defined here because if it was NEVER defined (i.e. undefined on first iteration),
-          // we would have returned in prev block, and if it was undefined on this iteration we would not overwrite
-          // that value.
-          // Tx was submitted: handle saving to store.
-          await this.handleTxSubmit(onchainTransactionId, method, methodId, channelAddress, reason, response!);
```

Comment:
So, working theory is that this line and the `response = tempResponse` line above were the problem lines.

Somehow, there was a response of `undefined` in a **subsuquent ** (2nd or greater number attempt) and we were *not* saving `tempResponse` to `response`... so we were calling `this.handleTxSubmit` and passing in a **stale response from the previous run**.

This resulted in uncaught prisma error - the txHash must be unique

Reply:
As to why we got `undefined` in a subsequent response, I don't know. @LayneHaber or @rhlsthrm maybe you can help me out with that. @LayneHaber I think there was some reason for it... maybe re-explain. This might be worth writing a unit test for that edge case